totalRP3/Core/Player.lua

```diff
@@ -450,6 +450,25 @@ function CurrentUser:SetCustomRace(race)
 	UpdateProfileField(self, "characteristics", "RA", race);
 end
 
+function CurrentUser:GetRoleplayExperience()
+	return TRP3_API.configuration.getValue("roleplay_experience");
+end
+
+function CurrentUser:SetRoleplayExperience(experience)
+	-- RP experience field is a bit special in that it's stored globally, but
+	-- transmitted as part of character data - so it must trigger a version
+	-- update and register-related events but can't go through the standard
+	-- UpdateProfileField path to do so.
+
+	local character = self:GetInfo("character");
+	local playerName = TRP3_API.globals.player_id;
+	local profileID = self:GetProfileID();
+
+	TRP3_API.configuration.setValue("roleplay_experience", experience);
+	character.v = TRP3_API.utils.math.incrementNumber(character.v or 1, 2);
```

Comment:
I'm not sure there's a good way around it, but incrementing only the current profile means if someone sees you on another profile instead, they won't get the updated status. Probably unlikely to happen often anyway, just thinking out loud.

Reply:
Hmm, that's a good point. Not entirely opposed to just incrementing the versions across *all* profiles, but will think.